### PR TITLE
Feat/add solidjs support

### PR DESCRIPTION
**Pull request by yisacc**
### Description

Add solidJS support implemented solid package to enable solid reactivity when web3-onboard is used with in a solidjs codebase.

### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### Docs Checklist
- [x] Include a screenshot of any changes ([see docs README on running locally](https://github.com/blocknative/web3-onboard/blob/develop/docs/README.md))
- [x] Add/update the appropriate package README (if applicable)
- [x] Add/update the related module in the [docs demo](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/lib/services/onboard.js) (if applicable)
- [x] Add/update the related package in the `docs/package.json` file (if applicable)

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (injected)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

#### Tests with demo app (SDK)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

### CODE REVIEW

File: packages/solid/package.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{
+  "name": "@web3-onboard/solid",
+  "version": "2.7.10",
```
Comment (Adamj1232) [inline packages/solid/package.json:3]:
```suggestion
  "version": "2.0.0-alpha.1",
```
We will set the initial published staging package to 2.0.0 with an alpha flag to signify a prerelease for published testing of this package.

**Comment by Adamj1232** [comment]
Thanks for so much for the PR to add SolidJs support! Any chance you could add a small example here in our examples section?
If so I will get this tested and published along with getting a section added to our docs as well.
You can also remove the pnpm lock file when you have a moment.
Thanks again!

**Review by bitcoinfinancier — APPROVED** [review]

**Review by bitcoinfinancier — APPROVED** [review]

**Comment by Adamj1232** [comment]
@yisacc we would love to get this merged - can you take a look at the failing PR checks and add a small boilerplate example to the examples folder when you have a moment?

**Comment by yisacc** [comment]
> @yisacc we would love to get this merged - can you take a look at the failing PR checks and add a small boilerplate example to the examples folder when you have a moment?

sure I will update it

**Comment by yisacc** [comment]
> @yisacc we would love to get this merged - can you take a look at the failing PR checks and add a small boilerplate example to the examples folder when you have a moment?

@Adamj1232 please do review and merge it

**Review by Adamj1232 — APPROVED** [review]

**Comment by Adamj1232** [comment]
@yisacc does solidjs require node version 18? The requirement in the package.json is causing issues with our build system as our minimum requirement is node v16

**Comment by Adamj1232** [comment]
@yisacc I am still seeing the PR checks fail because of the node versioning. Please see https://github.com/blocknative/web3-onboard/actions/runs/6538139277/job/17753445603?pr=1886

**Comment by yisacc** [comment]
> @yisacc I am still seeing the PR checks fail because of the node versioning. Please see https://github.com/blocknative/web3-onboard/actions/runs/6538139277/job/17753445603?pr=1886

@Adamj1232 solved

**Comment by Adamj1232** [comment]
@yisacc still seeing errors on the PR check

**Comment by yisacc** [comment]
> @yisacc still seeing errors on the PR check

@Adamj1232 i am working on it trying to solve the issues

**Comment by yisacc** [comment]
> @yisacc still seeing errors on the PR check

Now it's solved

**Comment by Adamj1232** [comment]
@yisacc thanks for that! I am doing some final testing and will add to our docs site.
Can you add an super simple example to the `/examples/` within the repo with a connect btn?

**Comment by yisacc** [comment]
> @yisacc thanks for that! I am doing some final testing and will add to our docs site. Can you add an super simple example to the `/examples/` within the repo with a connect btn? If you can add this I can get this package out in the release today/tomorrow! This is specifically helpful if webpack or some other bundler configs are required

@Adamj1232 i have added example

**Pull request merged**